crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -187,31 +195,113 @@ impl C8YHttpProxyActor {
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
         let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let req_build_closure = |jwt_token: String, _internal_id: String| {
+            let request_internal_id = HttpRequestBuilder::get(url_get_id.clone());
+            request_internal_id.bearer_auth(jwt_token)
+        };
```

Comment:
Its really unusual to name a closure like this.

Using a `move` you should be able to get rid of the `clone`.

Reply:
I named it to be `bit meaningful`. 

The `move` did not work in this case here, I was getting `expected a closure that implements the `Fn` trait, but this closure only implements `FnOnce` `. But I could get rid of the `clone` by capturing the `url_get_id` using reference.